drivers/retained_mem/retained_mem_rv3028.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2024 ANITRA system s.r.o.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT microcrystal_rv3028_retmem
+
+#include <zephyr/drivers/rtc/rtc_rv3028.h>
+
+LOG_MODULE_DECLARE(rv3028, CONFIG_RTC_LOG_LEVEL);
+
+#define RV3028_USER_RAM_SIZE (2 * sizeof(uint8_t))
+
+struct rv3028_retained_mem_config {
+	const struct device *parent;
+};
+
+static ssize_t rv3028_retained_mem_size(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return (ssize_t)RV3028_USER_RAM_SIZE;
+}
+
+static int rv3028_retained_mem_read(const struct device *dev, off_t offset, uint8_t *buffer,
+				    size_t size)
+{
+	const struct rv3028_retained_mem_config *config = dev->config;
+	const struct device *rtc_dev = config->parent;
+	int err;
+
+	rv3028_lock_sem(rtc_dev);
```

Comment:
There should be mutexes here (which are only present and used if `CONFIG_RETAINED_MEM_MUTEXES` is enabled) to guard against different threads calling these functions are the same time. See implementation in https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/retained_mem/retained_mem_nrf_gpregret.c for details

Reply:
Oh, I see. I’ll remove the semaphore that guards the RTC. However, I think an additional mutex is redundant since the underlying I2C driver must prevent simultaneous transactions.

Reply:
The issue with this is that it should be usable from a bootloader which does not have multithreading, that's what the `CONFIG_RETAINED_MEM_MUTEXES` Kconfig being disabled is for, if the i2c driver forces use of a mutex/semaphore then it's out of the scope

Reply:
I’ve removed semaphore locking from the retained memory interface implementation and am relying on the I2C API calls to be thread-safe. The dependent RTC driver still requires `k_sem`. Ensuring that the RTC driver works before the kernel is fully initialized is beyond the scope of this PR.